rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -1062,43 +1074,50 @@ protected ResourceManager registerInRM(final Credentials credentials, final Stri
 
-        // initializing JMX server with Sigar beans
-        SigarExposer sigarExposer = new SigarExposer(nodeName);
-        final RMAuthentication rmAuth = auth;
-        sigarExposer.boot(auth, false, new PermissionChecker() {
-            @Override
-            public boolean checkPermission(Credentials cred) {
-                ResourceManager rm = null;
-                try {
-                    rm = rmAuth.login(cred);
-                    if (NB_OF_ADD_NODE_ATTEMPTS == 0)
+        if(!disabledMonitoring) {
+            // initializing JMX server with Sigar beans
+            SigarExposer sigarExposer = new SigarExposer(nodeName);
+            final RMAuthentication rmAuth = auth;
+            sigarExposer.boot(auth, false, new PermissionChecker() {
+                @Override
+                public boolean checkPermission(Credentials cred) {
+                    ResourceManager rm = null;
+                    try {
+                        rm = rmAuth.login(cred);
+                        if (NB_OF_ADD_NODE_ATTEMPTS == 0)
+                            return true;
+
+                        boolean isAdmin = rm.isNodeAdmin(nodes.get(0).getNodeInformation().getURL())
+                                .getBooleanValue();
+                        if (!isAdmin) {
+                            throw new SecurityException("Permission denied");
+                        }
                         return true;
-
-                    boolean isAdmin = rm.isNodeAdmin(nodes.get(0).getNodeInformation().getURL())
-                            .getBooleanValue();
-                    if (!isAdmin) {
-                        throw new SecurityException("Permission denied");
-                    }
-                    return true;
-                } catch (LoginException e) {
-                    throw new SecurityException(e);
-                } finally {
-                    if (rm != null) {
-                        rm.disconnect();
+                    } catch (LoginException e) {
+                        throw new SecurityException(e);
+                    } finally {
+                        if (rm != null) {
+                            rm.disconnect();
+                        }
                     }
                 }
+            });
+
+            for (final Node node : nodes) {
+                try {
+                    node.setProperty(JMX_URL + JMXTransportProtocol.RMI, sigarExposer.getAddress(
+                            JMXTransportProtocol.RMI).toString());
+                    node.setProperty(JMX_URL + JMXTransportProtocol.RO, sigarExposer.getAddress(
+                            JMXTransportProtocol.RO).toString());
+                } catch (Exception e) {
+                    logger.error("", e);
+                }
             }
-        });
+        }else {
+            logger.info("JMX monitoring is disabled.");
+        }
 
         for (final Node node : nodes) {
```

Comment:
In a performance point of view, and in the case there are several nodes, it could have been better to let this for loop out of the disabledMonitoring condition (like it currently is), but add a if (!disabledMonitoring)...  Inside. To be discussed/confirmed..